colorpair.c

```diff
@@ -0,0 +1,35 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include "colorpair.h"
+
+int GetPairNumberFromColor(const ColorPair* colorPair) {
+    return colorPair->majorColor * numberOfMinorColors + colorPair->minorColor + 1;
+}
+
+void testPairToNumber(enum MajorColor major, enum MinorColor minor, int expectedPairNumber)
```

Comment:
production code and test code should be kept separately, so that test code can be excluded during software shipment to field. It also helps in maintenance.